### PR TITLE
[exotica_examples] Renaming and code standardization

### LINKED ISSUE #463: Dependency on Joy package
`PythonPlanIKEffAxisAlignment.launch`, `PythonPlanIKPointToLine.launch`, `PythonPlanIKWithJntSmooth.launch` depend on a physical joystick being plugged in and it has to be a very specific type of joystick.
These examples are therefore impossible to maintain without the hardware. 
All of these examples can either use generated input or use keyboard.

### PR DESCRIPTION
**Pull request by VladimirIvan**
- Moved examples from `examples/exotica_examples` to `exotica_examples`.
- Renamed all files consistently (including tests).
- Fixed broken examples (including workaround for #456).
- Removed dependency on joy and replaced it with an interactive marker (fixes #463).
- Visually checked all examples.
- Added python module with `TargetMarker` helper class for setting up interactive markers (used in some IK examples now).
- Updated the `run_examples` script to run all existing examples.

Notes:
 - `python_prm` runs very slowly.

### CODE REVIEW

File: exotica_examples/CMakeLists.txt [hunk 1]
```diff
@@ -51,13 +44,13 @@ install(DIRECTORY resources/ DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/res
 FILE(GLOB python_examples "${CMAKE_CURRENT_SOURCE_DIR}/scripts/*")
 catkin_install_python(PROGRAMS ${python_examples} DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
 
+catkin_python_setup()
```
Comment (wxmerkt) [inline exotica_examples/CMakeLists.txt]: I always thought this has to come _before_ `catkin_package()` but now I can't find this in the documentation anymore (cf. [here](http://docs.ros.org/melodic/api/catkin/html/user_guide/setup_dot_py.html))

File: exotica_examples/resources/rviz.rviz [hunk 2]
```diff
@@ -186,6 +188,6 @@ Window Geometry:
     collapsed: false
   Views:
     collapsed: false
-  Width: 1855
-  X: 1025
-  Y: 536
+  Width: 3738
```
Comment (wxmerkt) [inline exotica_examples/resources/rviz.rviz]: You now need a 4K screen for our examples, duh.

File: exotica_examples/scripts/example_prm [hunk 3]
```diff
@@ -12,6 +12,11 @@
 # You have to call prm.clear_query() between runs.
 # Call prm.clear() to discard the roadmap.
 
+# Temporary fix: call solve() to setup the OMPL problem,
```
Comment (wxmerkt) [inline exotica_examples/scripts/example_prm]: Add a reference to the issue

File: exotica_examples/src/exotica_examples_py/target_marker.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+from interactive_markers.interactive_marker_server import *
+from visualization_msgs.msg import *
+from geometry_msgs.msg import Pose, Point, Quaternion
+import PyKDL as kdl
+import pyexotica as exo
+
```
Comment (wxmerkt) [inline exotica_examples/src/exotica_examples_py/target_marker.py]: Add an `__all__ = [list_to_pose, TargetMarker]` here - so when people do `from exotica_examples_py import *` they do not get transient access to PyKDL etc.

File: exotica_examples/package.xml [hunk 5]
```diff
@@ -20,10 +20,11 @@
...
   <exec_depend>exotica_time_indexed_rrt_connect_solver</exec_depend>
-  <exec_depend>joy</exec_depend>
   <exec_depend>robot_state_publisher</exec_depend>
   <exec_depend>rviz</exec_depend>
-  <exec_depend>joy</exec_depend>
+  <exec_depend>geometry_msgs</exec_depend>
+  <exec_depend>visualization_msgs</exec_depend>
+  <exec_depend>interactive_markers</exec_depend>
   <test_depend>rostest</test_depend>
   <test_depend>exotica_val_description</test_depend>
```
Comment (wxmerkt) [inline exotica_examples/package.xml]: PyKDL dependency is missing

**Comment by wxmerkt** [comment]
The look_at example doesnt work for me though hmmm it's only moving the first two joints

File: exotica_examples/CMakeLists.txt (hunk 1 above)
Comment (VladimirIvan) [reply exotica_examples/CMakeLists.txt]: The docs say `When using the genmsg macro, ordering constraints exist`
and they provide this order:
```
project(...)
...
find_package(catkin ...)
...
catkin_python_setup()
...
generate_messages()
...
catkin_package()
```

The auto-generated CMakeLists also has it in that place.

**Comment by VladimirIvan** [comment]
> The look_at example doesnt work for me though hmmm it's only moving the first two joints

That's how it was designed to work. I don't like the look_at and the point_to_line examples for the same reason. They don't show what the task map does in an intuitive way. @cmower

File: exotica_examples/CMakeLists.txt (hunk 1 above)
Comment (wxmerkt) [reply exotica_examples/CMakeLists.txt]: Thanks, I knew I read it somewhere but couldnt find the reference!

**Comment by wxmerkt** [comment]
Hmmm, then the look_at name is a bit misleading (and not what a gaze, the alternative name suggestion, would do).

**Pull request merged**